src/main/java/org/mule/extension/sftp/internal/connection/MuleSftpClient.java

```diff
@@ -0,0 +1,178 @@
+    }
+
+    IoConnectFuture connectingFuture = connector.connect(targetAddress, context, localAddress);
+    connectFuture.addListener(c -> {
+      if (c.isCanceled()) {
+        connectingFuture.cancel();
+      }
+    });
+    connectingFuture.addListener(listener);
+    return connectFuture;
+  }
+
+
+  protected SshFutureListener<IoConnectFuture> createConnectCompletionListener(ConnectFuture connectFuture, String username,
+                                                                               InetSocketAddress address,
+                                                                               HostConfigEntry hostConfig) {
+    return new SshFutureListener<IoConnectFuture>() {
+
+      @Override
+      @SuppressWarnings("synthetic-access")
+      public void operationComplete(IoConnectFuture future) {
+        if (future.isCanceled()) {
+          CancelFuture cancellation = connectFuture.cancel();
+          if (cancellation != null) {
+            future.getCancellation().addListener(f -> cancellation.setCanceled(f.getBackTrace()));
+          }
+          return;
+        }
+
+        Throwable t = future.getException();
+        if (t != null) {
+          if (log.isDebugEnabled()) {
+            log.debug("operationComplete({}@{}) failed ({}): {}",
+                      username, address, t.getClass().getSimpleName(), t.getMessage());
+          }
+          connectFuture.setException(t);
+        } else {
+          IoSession ioSession = future.getSession();
+          try {
+            MuleSftpClientSession session = createSession(ioSession,
+                                                          username, address, hostConfig);
+            connectFuture.setSession(session);
+          } catch (RuntimeException e) {
+            connectFuture.setException(e);
+            ioSession.close(true);
+          }
+        }
+      }
+
+      @Override
+      public String toString() {
+        return "ConnectCompletionListener[" + username + "@" + address + "]";
+      }
+    };
+  }
+
+  private InetSocketAddress configureProxy(HostConfigEntry hostConfig) {
+    InetSocketAddress proxyAddress = new InetSocketAddress(proxyConfig.getHost(), proxyConfig.getPort());
+    InetSocketAddress remoteAddress = new InetSocketAddress(hostConfig.getHost(), hostConfig.getPort());
+
+    if (proxyAddress.isUnresolved()) {
+      proxyAddress = new InetSocketAddress(proxyAddress.getHostName(),
+                                           proxyAddress.getPort());
+    }
+
+    switch (proxyConfig.getProtocol()) {
+      case HTTP:
+        setClientProxyConnector(proxyConfig.getUsername() != null && proxyConfig.getPassword() != null
+            ? new HttpClientConnector(proxyAddress, remoteAddress, proxyConfig.getUsername(),
+                                      proxyConfig.getPassword().toCharArray())
+            : new HttpClientConnector(proxyAddress, remoteAddress));
+        break;
```

Comment:
We can remove the null checks since the else from the if-else clause uses a constructor that sends both parameters as null.
This applies for both socks5 and http
```suggestion
        setClientProxyConnector(new HttpClientConnector(proxyAddress, remoteAddress, proxyConfig.getUsername(),proxyConfig.getPassword().toCharArray())
        break;
```

Reply:
Sure!